lib/FilesHooks.php

```diff
@@ -499,7 +513,30 @@ protected function generateMoveActivities($users, $beforePathMap, $afterPathMap,
 	 * @return array
 	 */
 	protected function getUserPathsFromPath($path, $uidOwner) {
-		return Share::getUsersSharingFile($path, $uidOwner, true, true);
+		try {
+			$node = $this->rootFolder->getUserFolder($uidOwner)->get($path);
+		} catch (NotFoundException $e) {
+			return [];
+		}
+
+		if (!$node instanceof Node) {
+			return [];
+		}
+		$accessList = $this->shareManager->getAccessList($node, true, true);
+		$helper = new ShareHelper($this->rootFolder); // FIXME
+
+		$nodes = $helper->getPathsForAccessList($node, $accessList['users']);
```

Comment:
This sets up all the users and is therefor very slow :/
Also I wonder about a potential memory issue with nodes being loaded for all users